### PR TITLE
Issue 9 - Minification task

### LINKED ISSUE #9: Add minification gulp task
Minify html, css, js and images. For the images check if exists any gulp plugin with good compression, if not use the api of https://tinypng.com/

### PR DESCRIPTION
**Pull request by rodrijuarez**
This resolves #9 .

### CODE REVIEW

File: generators/gulp/templates/tasks/minifyCss.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import gulpif from 'gulp-if'
+import minifyCss from 'gulp-minify-css'
+import sourcemaps from 'gulp-sourcemaps'
+
+export default (gulp, options) => {
+    gulp.task('minify-css', function() {
+      return gulp.src('./src/client/assets/css/*.css')
```
Comment (beogip) [inline generators/gulp/templates/tasks/minifyCss.js]: Can we move the path to const options in gulpfile.babel.js? If we can, move all the paths of src to this file

File: generators/gulp/templates/tasks/minifyImages.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import imagemin from 'gulp-imagemin'
```
Comment (beogip) [inline generators/gulp/templates/tasks/minifyImages.js]: We need to implement the api of https://tinypng.com/developers. I test the compression of gulp-imagemin and compare it with tinypng. The initial size of the file was 2.2mb

gulp-imagemin
screen shot 2015-12-30 at 9 24 54 pm
Tinypng
screen shot 2015-12-30 at 9 24 28 pm

Comment (rodrijuarez) [inline generators/gulp/templates/tasks/minifyImages.js]: In the next commits I've used a gulp module found on the internet which uses the tinypng API, probably we will need to change the key in the future and upgrade the plan (after 500 images compression each image costs money), if the images are not in the format that tinypng requires, we use the old minification

**Comment by beogip** [comment]
Removed tinypng key from code, added config.json, fixed tinypng environment checker, added src/client/assets structure

**Pull request merged**